pkg/cacheutil/memcached_client.go

```diff
@@ -177,6 +182,11 @@ type memcachedClient struct {
 	dataSize   *prometheus.HistogramVec
 }
 
+type AddressProvider interface {
```

Comment:
Could you please provide a comment for this interface:
```
// AddressProvider is ...
```

Also, `dns.Provider` should implement it, right? We could write: `var _ AddressProvider = (*Provider)(nil)`